### ISSUE TITLE
[Feature Request] concat shape support for ViT

### ISSUE THREAD

**Issue by jliangTT** [tenstorrent-metal/tt-metal#7270]
**Is your feature request related to a problem? Please describe.**

Concat is currently falling back to CPU and is killing e2e perf. 

See: https://tenstorrent.sharepoint.com/:x:/r/sites/Jasmina/_layouts/15/Doc.aspx?sourcedoc=%7B4D45EB45-E309-4B38-883D-CEC7D4CD93FB%7D&file=ops_perf_results_vit_E2E__2024_04_04_21_25_15.xls&action=default&mobileredirect=true 

**Describe the solution you'd like**
Concat Support for this configuration per ViT.

**Describe alternatives you've considered**

**Additional context**

**Comment by tarafdarTT** [comment]
@mbahnasTT

**Comment by tarafdarTT** [comment]
checked with @mbahnasTT's unit test. I added it here on my own branch:
https://github.com/tenstorrent-metal/tt-metal/commit/3cd19c6049ff65391540be0d53527fc412bb3e5b


TTNN for some reason is forcing a fallback if it's not tilized, in my commit I hacked ttnn to let this through, and it works on device. @arakhmati @eyonland will need to see why @mbahnasTT 's cases are fallback since it does work on device.
